application/src/main/kotlin/net/corda/v5/application/messaging/FlowMessaging.kt

```diff
@@ -8,12 +9,38 @@ import net.corda.v5.base.types.MemberX500Name
 interface FlowMessaging {
 
     /**
-     * Creates a communication session with [party]. Subsequently you may send/receive using this session object. Note
-     * that this function does not communicate in itself, the counter-flow will be kicked off by the first send/receive.
+     * Creates a communication session with another member. Subsequently you may send/receive using this session object.
+     * Note that this function does not communicate in itself, the counter-flow will be kicked off by the first
+     * send/receive.
+     *
+     * Initiated flows are initiated with context based on the context of the initiating flow. The context of the
+     * initiating flow is snapshotted by the returned session when this method is called. Altering the parent context
+     * has no effect on the context of the session after this point, and therefore it has no effect on the context of
+     * the initiated flow either.
+     *
+     * @param x500Name The X500 name of the member to communicate with.
+     * @return The session.
      */
     @Suspendable
     fun initiateFlow(x500Name: MemberX500Name): FlowSession
 
+    /**
+     * Creates a communication session with another member. Subsequently you may send/receive using this session object.
+     * Note that this function does not communicate in itself, the counter-flow will be kicked off by the first
+     * send/receive.
+     *
+     * This overload takes a mutator of context properties. Any properties set or modified against the context passed to
+     * this mutator will be propagated to initiated flows and all that flow's initiated flows and sub flows down the
+     * stack. The properties passed to the mutator are pre-populated with the parent context properties, see
+     * [FlowContextProperties].
+     *
+     * @param x500Name The X500 name of the member to communicate with.
+     * @param flowContextPropertiesMutator A mutator of context properties
```

Comment:
Sorry to be pedantic:
Missing full stop on flowContextProperiesMutator
The style for the API KDocs is to have a new line between the sets of @param, @retrun and @throws